### DISCUSSION TITLE
subscriptions handled by graphql-core

### DISCUSSION THREAD

**Discussion by kristjanvalur** [strawberry-graphql/strawberry#2701] [General]
I noticed that in `strawberry.schema.schema`, `execute()` is imported from `strawberry.schema.execute` whereas `subscribe()` comes directly from graphql core.

strawberry `execute()` does various things, including running extensions, and returns a `strawberry.types.ExecutionResult` whereas the other does no strawberry things and returns `graphql.ExecutionResult`.

What are  we missing out by this?  What does `strawberry.schema.execute.execute` add on top of `graphql.execute()`?

The reason I ask is among other things that it causes a certain disparity between multi- and single-results operations over websockets. 

Indeed, there is a "TODO" in the schema, indicating that "extensions" will at some point be supported.  Is that all that strawberry adds on top, these extensions?  Should we add a similar strawberry wrapping for "subscribe()?"

**Comment by DoctorJohn** [comment]
Personally I didn't follow that issue too much but it has been discussed some times. Here is some context I remember from the top of my head:

There was some confusion in the past about how some of our extension hooks would work with subscriptions. That's because Strawberry had little subscription support when extensions were added. Not sure whether this confusion has been fully resolved yet.

However, there is a PR which adds extension support to subscriptions #2430. Originally it stalled for a bit since it changes 50+ files and was based on another PR. That other PR has been merged since then, so I'll take another look as soon as I find time.

**Reply by kristjanvalur** [reply]
Right, I don't know much about extensions myself, I only noticed the discrepancy in the internal api, with different "result" types being returned.  Which again is making type hints a bit tedious.

Briefly looking at `schema.execute` the code also mentions schema validation being done in different places.

I was just wondering if there was anything more being wrapped around the "core" execute by strawberry, which we are somehow missing out by not doing the same with "subscribe".  Otherwise I am not too worried about it :)
